judgels-frontends/raphael/src/components/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+import { stringify, parse } from 'query-string';
+import { push } from 'react-router-redux';
+import { withRouter, RouteComponentProps } from 'react-router';
+import { connect } from 'react-redux';
+
+import SearchBoxForm, { SearchBoxFormData } from './SearchBoxForm';
+
+interface SearchBoxRouteProps extends RouteComponentProps<{}> {
+  onAppendRoute: (queries: any) => any;
+}
+
+export interface SearchBoxProps {
+  onRouteChange: (content: string, prevQueries: any) => any;
+  initialValue?: string;
+  isLoading?: boolean;
+}
+
+const SearchBoxContainer = (props: SearchBoxProps & SearchBoxRouteProps) => {
```

Comment:
Let's add an integration test for this. Particularly, when the form is submitted, `onRouteChange` is called with the correct form value and queries. See `Pagination.test.tsx` for reference.